utils/include/LinuxUtils.cpp

```diff
@@ -81,6 +81,74 @@ namespace PlatformUtils
 		}
 		return false;
 	}
+
+	bool TryGetLogicalDisksNames(std::vector<std::string>& all_disks_names)
+	{
+		constexpr char path_dev[] = { "/dev" };
+		constexpr char loop_ignore[] = { "loop" };
+		std::vector<std::string> all_received_names;
+		//get all names of disks disks to mount
+		for (const auto& file : std::filesystem::directory_iterator(path_dev))
+		{
+			if (file.is_block_file() &&
+				(std::string::npos ==
+					file.path().filename().string().find(loop_ignore)))
+			{
+				all_received_names.emplace_back(
+					file.path().filename().string());
+			}
+		}
+		//mount all disks;
+		const std::string mount_dir = "/media/disks";
+		if (!std::filesystem::exists(mount_dir))
+		{
+			//make permision only /root 
+			if (!std::filesystem::create_directory(mount_dir))
+			{
+				return false;
+			}
+			std::filesystem::permissions(
+				mount_dir, std::filesystem::perms::group_all |
+				std::filesystem::perms::owner_all);
+		}
+		for (unsigned short i = 0; i < all_received_names.size(); ++i)
+		{
+			std::string path_to_mount_dir = mount_dir + "/" +
+				all_received_names[i];
+
+			if (!std::filesystem::exists(path_to_mount_dir))
+			{
+				if (!std::filesystem::create_directory(path_to_mount_dir))
+				{
+					return false;
+				}
+			}
+			//todo: try again make as API
+			//mount((path + "/" + all_mounted_disks_path[i]).c_str(), 
+			//(mount_dir + "/" + all_mounted_disks_path[i]).c_str(), 
+			//NULL, 0, NULL);
+
+			const std::string system_utils_mount_disk =
+				(static_cast<std::string>("mount ") + path_dev + "/" +
+					all_received_names[i] + " " + path_to_mount_dir);
+			system(system_utils_mount_disk.c_str());
+			system("clear");
+			//if mount to path_to_mount_dir FAIL, then delete name of 
+			//dir and delete created dir 
+			if (std::filesystem::is_empty(path_to_mount_dir))
+			{
+				//write to log
+				all_received_names.erase(all_received_names.begin() + i);
+				std::filesystem::remove(path_to_mount_dir);
+				continue;
+			}
+
+			all_disks_names.emplace_back(path_to_mount_dir);
+		}
+
+		return !all_disks_names.empty();
+	}
+
```

Comment:
We need to split this function at least to: Mount() and GetNames().

Reply:
Right now I'm implementing getting names from dir /dev in another function. In a close future, I move directory creating stuff to another function, but it will take some time. Will be done in the next demo